### PR TITLE
Notifications and integration of profile using context

### PR DESCRIPTION
**Pull request by hiteshparate**
### What this PR does (required):
- Integration of profile and getting notification from real data

### Screenshots / Videos (required):
- https://user-images.githubusercontent.com/48521519/123686013-52383a80-d81d-11eb-9bb4-83e5966b0463.mp4

### CODE REVIEW

File: client/src/context/useAuthContext.tsx [hunk 1]
```diff
@@ -4,27 +4,44 @@ import { AuthApiData, AuthApiDataSuccess } from '../interface/AuthApiData';
...
   const updateLoginContext = useCallback(
     (data: AuthApiDataSuccess) => {
       setLoggedInUser(data.user);
+      getProfile(data.user.id).then((res) => {
+        setUserProfile(res.success?.profile);
+        getNotifications(res.success?.profile._id).then((result) => {
+          setNotifications(result.notifications);
+        });
+      });
+
```
Comment (ajaytomgeorge) [inline client/src/context/useAuthContext.tsx:44]: Chaining requests too much would affect readability

File: server/controllers/profile.js [hunk 2]
```diff
@@ -1,77 +1,87 @@
...
-  if (!profiles) {
-    res.status(404);
-    throw new Error("Cannot access server.");
-  }
-  res.status(200).json({ profiles });
+exports.allProfiles = asyncHandler(async(req, res, next) => {
+    const profiles = await Profile.find();
+    if (!profiles) {
+        res.status(404);
+        throw new Error('Cannot access server.');
```
Comment (ajaytomgeorge) [inline server/controllers/profile.js:27]: If there are no profiles, it should return an empty array, you can return it, rather Missing Resource

**Comment by bonnieli** [comment]
Hi @hiteshparate - please update your branch here to resolve the merge conflicts. Hopefully that would update your branch to reflect the right amount of files and changes as well

File: client/src/components/Notification/Notification.tsx [hunk 3]
```diff
@@ -1,41 +1,26 @@
...
 
   return (
     <>
-      {console.log(notifications)}
       {notifications &&
         notifications.map((notification, index) => (
           <Box p={1} key={index}>
             <Button>
-              {notification.sender} has {notification.type}ed your service
+              {notification.sender === null ? 'unknown' : notification.sender.firstName} has {notification.type}ed your
```
Comment (rajivtitus) [inline client/src/components/Notification/Notification.tsx:14]: As opposed to hardcoding this text, you can display a message based on the type of notification (message, request).

File: server/controllers/notifications.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ const asyncHandler = require('express-async-handler');
 const Notifications = require('../models/Notifications');
 
 exports.getAllNotifications = asyncHandler(async(req, res) => {
-    const receiverProfileId = req.body.profileId;
+    const receiverProfileId = req.params.id;
```
Comment (rajivtitus) [inline server/controllers/notifications.js:5]: You can maybe refactor controllers to get notifications with the user's id (request object).
Eg: `req.user.id`

File: server/controllers/notifications.js [hunk 5]
```diff
@@ -39,8 +39,8 @@ exports.createNotification = (async(req, res) => {
 });
 
 exports.getUnreadNotifications = (async(req, res) => {
-    const receiverProfileId = req.body.profileId;
-    const notifications = await Notifications.find({ receiver: receiverProfileId, read: false });
+    const receiverProfileId = req.params.id;
```
Comment (rajivtitus) [inline server/controllers/notifications.js:42]: Same as above :)

File: server/controllers/profile.js [hunk 6]
```diff
@@ -1,77 +1,87 @@
...
 });
 
 // @route PATCH /profile/:id
 // Edit a profile with a matching id
 
-exports.editProfile = asyncHandler(async (req, res, next) => {
-  const id = req.params.id;
-  const profile = req.body;
+exports.editProfile = asyncHandler(async(req, res, next) => {
+    const id = req.params.id;
```
Comment (rajivtitus) [inline server/controllers/profile.js:47]: Please refactor route as described above (Notifications route)

**Review by rajivtitus — COMMENTED** [review]
Please have a look at the comments and address. Thanks :)

**Pull request merged**